### ISSUE TITLE
Request to add Fee ability to Shopping Cart Price Rule

### ISSUE THREAD

**Issue by robert198222** [magento/magento2#1573]
As standard in 1.x it's only possible to add Discounts as Actions in Shopping Cart Price Rule. It would be a natural future to have both Discounts and Fees as rules. Fees could be useful if your Condition is a Payment Method and you want to add a fixed fee or/and percent of product price fee for the chosen Payment Method, for example Paypal. Same if you want to add extra fee for Conditions with very heavy products, special destinations, if subtotal is too low etc.

**Comment by tkacheva** [comment]
Thanks! Added to product backlog MAGETWO-11611

**Comment by choukalos** [comment]
Hi @robert198222 I've gotten a few requests for add'l fees from a tax perspective and am jumping in here because I'm wondering if fees should really be part of the shopping cart price rule or part of some other functional areas.  Can you give examples of merchants/countries where they're charging add'l fees to use Paypal ( or other payment services )?  For conditions such as heavy products - how would you create the condition - would it be based off summation of attribute values ( aka weight ) and a threshold or off particular products (that are very heavy.... if this path then why not use FPT feature)?  If other scenarios - can you provide use cases?

I'm adding @asemenenko (payments PM) and @ilol (shipping PM) to track this issue as some of what you've brought up touches their domains and they might be tracking similar requests.

Thanks!

**Comment by robert198222** [comment]
Hi @choukalos there are several extensions for 1.x that provide additional fees for paypal and other payment options. example of the two most popular on Magento Connect: http://www.magentocommerce.com/magento-connect/paypal-payment-fee.html
http://www.magentocommerce.com/magento-connect/multi-fees-1.html

Several Payment providers (for example Paypal) charge store owners a percental fee. This would give store owners possibility to charge their customers with the same percental fee which they pay for the payment services so that they don't loose revenue because of the service that they provide. 

This condition I would set up like this:
If ALL  of these conditions are TRUE :
Payment Method  is  PayPal Website Payments Standard
(These conditions are copied directly from Magento)

And in Action I would set up like this:
Apply: Percent of product price Fee ("Percent of product price Fee" option doesn't exist but "Percent of product price Discount" exists.)

{For conditions such as heavy products - how would you create the condition}
In Conditions I would add following:
If ALL  of these conditions are TRUE :
Total Weight  equals or greater than  100  
Subtotal  less than  1000  
Shipping Country  is  Unites States
(These conditions are copied directly from Magento)

In action I would add an extra fee of 200 (As it is in 1.x it's only possible to give discount).

Basically, this would mean that if someone from United States tries to buy something for less than 1000USD but the weight is 100kg or more then there would be an extra fee of 200USD. Depending of shipment, several logistic companies such as DHL charge additional fees based on countries/postcodes/Regions/States/Provinces, this would give the opportunity for store owners to charge their customers same fee as they pay. Also, if you live in EU, you could easily set rules with extra fees to all countries outside EU.

Thanks!

**Comment by asemenenko** [comment]
@robert198222 , @tkacheva @choukalos , @ilol 

Answering your questions on legal aspect of these change:

In US it is allowed to apply a surcharge for at least Visa and Mastercard under certain conditions described in the respective documents, see:
https://www.mastercard.us/en-us/merchants/get-support/merchant-surcharge-rules.html
http://usa.visa.com/personal/get-help/checkout-fees.jsp

In other regions, say in UK, rules may be similar, below is an example of such rule explained in "Guidance on the Consumer Protection (Payment Surcharges) Regulations" that says:
> Q. Will all payment surcharges be prohibited?
> No. Traders will – if they wish – still be able to require payment surcharges but the amount
> will be limited to the costs they bear for accepting payments by the particular method of
> payment. So, if, for example, there is a surcharge for payment by credit card but not for
> other means of payment such as cash, the surcharge can be no higher than the additional
> costs the trader incurs in processing credit card payments. 

With this being said according to PayPal User Agreement (https://www.paypal.com/us/webapps/mpp/ua/useragreement-full):
> 4.5 No Surcharges. You agree that you will not impose a surcharge or any other fee for accepting > PayPal as a payment method. You may charge a handling fee in connection with the sale of goods > or services as long as the handling fee does not operate as a surcharge and is not higher than the > handling fee you charge for non-PayPal transactions.

My advice will be to always validate assumptions with the local law and against an agreement you have with your payment service provider.

**Comment by thomasnordkvist** [comment]
@robert198222 , @tkacheva , @choukalos , @ilol 

This would be a good feature for us.

It's common for B2B merchants to want to sneak in a fee for some kind of handling.

And also adding a fee if total cart is below some amount/quantity is quite common,
often used to compensate for orders with small values/quantities.

/Thomas

**Comment by khoatruongbc** [comment]
Hello all,
Any update this feature? I think we need to have a surcharge.

**Comment by ghost** [comment]
This would be a good feature to have to charge a Cash On Delivery fee for that payment method. For COD payments it could even be worked into the COD payment module to have an option to charge a fee.

**Comment by tkacheva** [comment]
@khoatruongbc , @crtsl 
Surcharge is in the scop of our promotion enhancements backlog. Follow the announcements of magento roadmap updates.

**Comment by piotrekkaminski** [comment]
Thank you for your submission.

We recently made some changes to the way we process GitHub submissions to more quickly identify and respond to core code issues.

Feature Requests and Improvements should now be submitted to the new [Magento 2 Feature Requests and Improvements](https://community.magento.com/t5/Magento-2-Feature-Requests-and/idb-p/feature-requests) forum (see details [here](https://community.magento.com/t5/News-Announcements/Improvements-to-GitHub-Management/m-p/44572#M96)).

We are closing this GitHub ticket and have moved your request to the new [forum](https://community.magento.com/t5/Magento-2-Feature-Requests-and/Request-to-add-Fee-ability-to-Shopping-Cart-Price-Rule/idi-p/46542).

**Comment by maxiiemaxx** [comment]
updates?
